### PR TITLE
fix: Update CODEOWNERS to match directory shuffle

### PR DESCRIPTION
**Pull request by martijneken**
Fix CODEOWNERS reviewers to cover plugins/ and callouts/ samples dirs.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,5 +2,6 @@
 *                 @GoogleCloudPlatform/service-extensions-samples-owners @GoogleCloudPlatform/dee-infra
 
 # Sample Reviewers
-samples/          @GoogleCloudPlatform/service-extensions-samples-reviewers @upeeth
+callouts/         @GoogleCloudPlatform/service-extensions-samples-reviewers @upeeth
```
Comment (mpwarres) [inline .github/CODEOWNERS:5]: Consider adding a leading '/' to `callouts/`, `plugins/`, and `README.md`, so that they match specifically at top-level.

**Review by mpwarres — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (martijneken) [reply .github/CODEOWNERS]: Done, good call!

**Comment by jstraceski** [comment]
If applicable, in a followup you can also add me to the callouts directory.

**Comment by martijneken** [comment]
> If applicable, in a followup you can also add me to the callouts directory.

You are already a member of service-extensions-samples-reviewers. Feel free to send PR to get explicitly added to reviews.

**Pull request merged**